From d15450ae2c7e18e031b0e9cf7fb623e2295ec897 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Wed, 28 Mar 2018 11:34:00 +0800
Subject: [PATCH] drm/rockchip: dw_hdmi: Fix up screen flash when switching
 color

After switching color, hdmi output signal may be unstable.
If AVMUTE is cleared too early, tv will display err.

Change-Id: I595180bfe6e014de5231bcd75ee259d5702121e0
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index 91a747d57cc7..9ac1c74c7efc 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -2667,6 +2667,11 @@ dw_hdmi_connector_atomic_flush(struct drm_connector *connector,
 
 	if (hdmi->hdmi_data.update) {
 		dw_hdmi_setup(hdmi, &hdmi->previous_mode);
+		/*
+		 * Before clear AVMUTE, delay is needed to
+		 * prevent display flash.
+		 */
+		msleep(50);
 		hdmi_writeb(hdmi, HDMI_FC_GCP_CLEAR_AVMUTE, HDMI_FC_GCP);
 		hdmi->hdmi_data.update = false;
 	} else if (connector->state->hdr_metadata_changed &&
-- 
2.35.3

